sklearn/metrics/ranking.py

```diff
@@ -98,26 +98,43 @@ def auc(x, y, reorder='deprecated'):
     if reorder != 'deprecated':
         warnings.warn("The 'reorder' parameter has been deprecated in "
                       "version 0.20 and will be removed in 0.22. It is "
-                      "recommended not to set 'reorder' and ensure that x "
-                      "is monotonic increasing or monotonic decreasing.",
+                      "recommended not to set 'reorder' and rest assured "
+                      "that x direction is internally taken into "
```

Comment:
that's kind of the opposite of what you're doing.... you're assuming the order is meaningful.

Reply:
Of course the direction is important & meaningful, but you do not need to force the user to have monotonic input. Just take care of the direction within smaller chunks, without annoying the user.